resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -255,13 +255,17 @@ private[spark] object KubernetesConf {
     s"spark-${UUID.randomUUID().toString.replaceAll("-", "")}"
 
   def getResourceNamePrefix(appName: String): String = {
-    val id = KubernetesUtils.uniqueID()
-    s"$appName-$id"
-      .trim
-      .toLowerCase(Locale.ROOT)
-      .replaceAll("[^a-z0-9\\-]", "-")
```

Comment:
how about replacing these characters with 'x', which makes us easily refer to unknown ones?

Reply:
The suggestion sounds good to me, which make the generated name easier to meet the K8s requirements, and if we adopt it, maybe `getAppNameLabel` should be updated as well.

will do it if no objections till tomorrow.